### PR TITLE
Netplan - WIP

### PR DESCRIPTION
**Pull request by jvonau**
### Fixes Bug
aimed at #1320 
WIP

### CODE REVIEW

File: roles/network/templates/network/dnsmasq.service.u18 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Unit]
```
Comment (holta) [inline roles/network/templates/network/dnsmasq.service.u18]:
```suggestion
[Unit]
```

File: roles/network/tasks/netplan.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  when: wan_ip != "dhcp"
+
+- name: Static IP setting CIDR
+  set_fact:
+    wan_cidr: "{{ CIDR.stdout }}"
+  when: wan_ip != "dhcp"
+
+- name: Supply netplan template
+  template:
+    dest:  /etc/netplan/01-iiab-config
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
    dest: /etc/netplan/01-iiab-config
```

File: roles/network/tasks/netplan.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    src: network/netplan.j2
+    backup: no
+
+- name: generate netplan config
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
- name: Generate netplan config
```

File: roles/network/tasks/netplan.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  shell: netplan generate --debug
+
+- name: Stopping services
+  include_tasks: down-debian.yml
+
+# wants a controlling terminal for the ENTER key, so it fails
+- name: test netplan config
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
- name: Test netplan config
```

**Comment by tim-moody** [comment]
I certainly like the sound of this. It's what I wanted to write, a service that configures at boot to what is wanted, hopefully degrading to something usable otherwise.

File: roles/network/tasks/netplan.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
+# netplan.yml
+- name: Figure out netplan file name on Ubuntu 18
+  shell: ls /etc/netplan
+  register: netplan
+
+- name: Disable cloud-init the easy way
+  shell: touch /etc/cloud/cloud-init.disabled
+  when: ("item" == "50-cloud-init.yaml")
+  with_items:
+   - "{{ netplan.stdout }}"
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
    - "{{ netplan.stdout }}"
```

File: roles/network/tasks/netplan.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+- name: Remove stock netplan template
+  file:
+    state: absent
+    dest: /etc/netplan/{{ item }}
+  with_items:
+   - "{{ netplan.stdout }}"
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
    - "{{ netplan.stdout }}"
```

File: roles/network/tasks/netplan.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    src: network/dnsmasq.service.u18
+    dest: /etc/systemd/system/iiab-dnsmasq.service
+#  when: systemd_networkd_active
+
+- name: Ensure required dnsmasq unit files are enabled for reboot systemd_networkd_active
+  systemd:
+    name: "{{ item }}"
+    enabled: yes
+  with_items:
+   - iiab-dnsmasq
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
    - iiab-dnsmasq
```

File: roles/network/tasks/netplan.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+   - systemd-networkd-wait-online
```
Comment (holta) [inline roles/network/tasks/netplan.yml]:
```suggestion
    - systemd-networkd-wait-online
```

**Comment by holta** [comment]
To be tested on Ubuntu 18.04.1 and Raspbian-on-RPi 3 then hopefully merged soon in coming days.

Thanks @jvonau

Unrelated but also likely to make the final cut for IIAB 6.7:
#234 -> PR #1279 for #1237

**Comment by holta** [comment]
I'll try testing this on various machines in the coming ~24h — thanks all for reviewing this prior to merging this PR tmrw hopefully?

**Comment by holta** [comment]
PR1336 seems to exhibit 2 failures on Raspbian Desktop:
- "systemctl status hostapd" shows hostapd is failing to start
- when hostapd is manually started/enabled (as @jvonau did last night on 10.8.0.66) WiFi devices cannot connect to the AP/SSID, despite this AP's radio (WiFi beaconing) being very strong

**Comment by holta** [comment]
> 
> 
> PR1336 seems to exhibit 2 failures on Raspbian Desktop:
> 
>     * "systemctl status hostapd" shows hostapd is failing to start
> 
>     * when hostapd is manually started/enabled (as @jvonau did last night on 10.8.0.66) WiFi devices cannot connect to the AP/SSID, despite this AP's radio (WiFi beaconing) being very strong

CLAIRF from @jvonau: above is a non-issue, which was caused by the WiFi being a gateway, locking out AP functionality.

All that should be needed is replacing `dhcpd` with `dnsmasq` in both scripts **[iiab-hotspot-on | off](https://github.com/iiab/iiab/tree/master/roles/network/templates/network)**.

**Comment by holta** [comment]
Tested on Raspian Desktop on RPi 3 B+ & Ubuntu Server 18.04.1 on NUC with @jvonau's help.

On rebooting each IIAB above:
- "systemctl status hostapd dnsmasq" shows both services running successfully
- connecting a separate WiFi device to http://box works
- unfortunately, Captive Portal does not yet work with Android 8.1 client, reconfirming https://github.com/iiab/iiab/issues/1323 :/

**Pull request merged**